src/humancellatlas/data/metadata/helpers/date_and_time.py

```diff
@@ -0,0 +1,43 @@
+from datetime import (
+    datetime,
+    timezone,
+)
+from typing import (
+    Optional,
+)
+
+from dateutil import (
+    parser,
+)
+
+
+def datetime_from_string(s: str) -> Optional[datetime]:
+    """
+    Convert a string into a tz-aware (UTC) datetime
+
+    >>> datetime_from_string('2019-09-20T082959.187000Z')
+    datetime.datetime(2019, 9, 20, 8, 29, 59, 187000, tzinfo=datetime.timezone.utc)
+
+    >>> datetime_from_string('2018-10-07T12:00:39.740Z')
+    datetime.datetime(2018, 10, 7, 12, 0, 39, 740000, tzinfo=datetime.timezone.utc)
+
+    >>> datetime_from_string('2021-05-05T14:24:26.174274-07:00')
+    datetime.datetime(2021, 5, 5, 21, 24, 26, 174274, tzinfo=datetime.timezone.utc)
+
+    >>> datetime_from_string('November 1, 1974 11:11:11 PST')
+    datetime.datetime(1974, 11, 1, 19, 11, 11, tzinfo=datetime.timezone.utc)
+
+    >>> datetime_from_string(None)
+    >>> str(datetime_from_string(''))
+    Traceback (most recent call last):
+    ...
+    dateutil.parser._parser.ParserError: String does not contain a date: \n
+    >>> datetime_from_string('today')
+    Traceback (most recent call last):
+    ...
+    dateutil.parser._parser.ParserError: Unknown string format: today
+    """
+    if s is None:
+        return None
+    d = parser.parse(s)
```

Comment:
This parses many more formats than what the schema allows, which is only [ISO-datetime](https://datatracker.ietf.org/doc/html/rfc3339#section-5.6). Python has built-in mechanisms for parsing that.